.infra/infra/main.tf

```diff
@@ -0,0 +1,270 @@
+  tag_id      = snowflake_tag.sensitivity[0].id
+  tag_value   = "PII"
+}
+
+// ORDERS TABLE
+resource "snowflake_table" "ordering_orders" {
+  database = snowflake_database.db.name
+  schema   = snowflake_schema.ordering.name
+  name     = "ORDERS"
+
+  column {
+    name = "ORDERKEY"
+    type = "NUMBER(38,0)"
+  }
+
+  column {
+    name = "CUSTKEY"
+    type = "NUMBER(38,0)"
+  }
+
+  column {
+    name = "ORDERSTATUS"
+    type = "VARCHAR"
+  }
+
+  column {
+    name = "TOTALPRICE"
+    type = "NUMBER(38,0)"
+  }
+
+  column {
+    name = "ORDERDATE"
+    type = "DATE"
+  }
+
+  column {
+    name = "ORDERPRIORITY"
+    type = "VARCHAR"
+  }
+
+  column {
+    name = "CLERK"
+    type = "VARCHAR"
+  }
+
+  column {
+    name = "SHIPPRIORITY"
+    type = "NUMBER(38,0)"
+  }
+
+  column {
+    name = "COMMENT"
+    type = "VARCHAR"
+  }
+}
+
+// SUPPLIER
+resource "snowflake_table" "ordering_supplier" {
+  database = snowflake_database.db.name
+  schema   = snowflake_schema.ordering.name
+  name     = "SUPPLIER"
+
+  column {
+    name = "SUPPKEY"
+    type = "NUMBER(38,0)"
+  }
+
+  column {
+    name           = "NAME"
+    type           = "VARCHAR"
+    masking_policy = var.snowflake_standard_edition ? null : snowflake_masking_policy.masking_policy[0].qualified_name
+  }
+
+  column {
+    name = "ADDRESS"
+    type = "VARCHAR"
+  }
+
+  column {
+    name = "NATIONKEY"
+    type = "NUMBER(38,0)"
+  }
+
+  column {
+    name = "PHONE"
+    type = "DATE"
+  }
+
+  column {
+    name = "ACCTBAL"
+    type = "VARCHAR"
+  }
+
+  column {
+    name = "COMMENT"
+    type = "VARCHAR"
+  }
+}
+
+resource "snowflake_masking_policy" "masking_policy" {
+  count = var.snowflake_standard_edition ? 0 : 1
+
+  name     = "ORDERING_MASKING_POLICY"
+  database = snowflake_database.db.name
+  schema   = snowflake_schema.ordering.name
+  signature {
+    column {
+      name = "val"
+      type = "VARCHAR"
+    }
+  }
+  masking_expression = <<-EOF
+    case
+      when current_role() in ('ACCOUNTADMIN', 'SYSADMIN') then
+        val
+      else
+        '******'
+    end
+  EOF
+
+  return_data_type = "VARCHAR"
+}
+
+resource "snowflake_tag_association" "supplier_pii" {
+  count = var.snowflake_standard_edition ? 0 : 1
+
+  object_identifier {
+    name     = "${snowflake_table.ordering_supplier.name}.ADDRESS"
+    database = snowflake_database.db.name
+    schema   = snowflake_schema.ordering.name
+  }
+
+  object_identifier {
+    name     = "${snowflake_table.ordering_supplier.name}.NAME"
+    database = snowflake_database.db.name
+    schema   = snowflake_schema.ordering.name
+  }
+
+  object_identifier {
+    name     = "${snowflake_table.ordering_supplier.name}.PHONE"
+    database = snowflake_database.db.name
+    schema   = snowflake_schema.ordering.name
+  }
+
+  object_type = "COLUMN"
+  tag_id      = snowflake_tag.sensitivity[0].id
+  tag_value   = "PII"
+}
+
+// ORDERS VIEW
+resource "snowflake_view" "orders_limited" {
```

Comment:
I propose that we make sure to have all types of tables and views (e.g. materialized and not)